### PR TITLE
Add integration test workflow

### PR DESCRIPTION
**Pull request by HakkyuKim**
Github workflows script that runs integration test on pull requests. This PR should be merged after setting up self-hosted runner.

Notes:
- Script doesn't run on forked repositories to silence no self-hosted runner error.
- Currently only a single self-hosted runner runs on the server machine, this may lag CI when multiple commits are submitted to the repo in a short time period. I have a few solutions in mind but need to verify some things first. Anyhow, it will be implemented in the next PR.
- Currently only supports testing on wearable-5.5 emulator.

Testing done on https://github.com/flutter-tizen/plugins/pull/228.

### CODE REVIEW

**Comment by bbrto21** [comment]
please, rebase first

**Comment by HakkyuKim** [comment]
We can now merge this after adding a self-hosted runner for this repository.

**Review by bwikbs — APPROVED** [review]
Oh! This is what I'm looking forward to personally! 🔥

**Comment by bbrto21** [comment]
> self-hosted runner for this repository.

@HakkyuKim  Is our self-hosted runner ready?

**Comment by HakkyuKim** [comment]
@bbrto21 I'm working on something else in our internal git at the moment, I'll bring it up next Monday!

**Comment by HakkyuKim** [comment]
I'm really sorry about the delay, setting up a server inside the company is a slow process...

**Comment by HakkyuKim** [comment]
To share the current process...I'm requesting one security approval after another. Apparently some urls are blocked by firewalls even with extern network, unfortunately GitHub is one of them.

**Comment by HakkyuKim** [comment]
The test server can now listen to Github workflows from an external network, https://github.com/HakkyuKim/plugins/pull/14/checks?check_run_id=3585756046. However, wearable emulators just randomly crash even when no app is running. Mobile and TV emulators can't even get booted. I'm planning to reinstall the host OS, file an issue to Tizen SDK team, or use another server machine.

**Comment by HakkyuKim** [comment]
Reinstalling the OS did not work, I've filed an issue to Tizen SDK team.

**Comment by HakkyuKim** [comment]
The test server finally works, @swift-kim could you grant me the permission to add self-hosted runner for this repo so I can test this out?

File: .github/workflows/integration_test.yml
Comment (swift-kim) [inline .github/workflows/integration_test.yml:8]: Why did you comment out? Please convert this PR to draft if the change is not ready for review yet. Also you don't need the quotation marks for the branch name as it doesn't contain whitespace characters.

Comment (HakkyuKim) [reply .github/workflows/integration_test.yml:8]: The comment out part was meant to serve in the next PR when the server supports multiple self-hosted runners, I left a TODO comment above but I can just remove it and add it when it becomes relevant. I'll remove the quotation marks in the next commit.

File: .github/workflows/integration_test.yml
Comment (WonyoungChoi) [inline .github/workflows/integration_test.yml:11]: It seems to be an unnecessary condition.

Comment (HakkyuKim) [reply .github/workflows/integration_test.yml:11]: I've removed push on master trigger for now, so yes, thanks for the find.

File: .github/workflows/integration_test.yml
Comment (WonyoungChoi) [inline .github/workflows/integration_test.yml:19]: how about using `actions/checkout`?
```yaml
      - uses: actions/checkout@v2
        with:
          repository: flutter-tizen/flutter-tizen
          path: flutter-tizen
```

File: .github/workflows/integration_test.yml
Comment (WonyoungChoi) [inline .github/workflows/integration_test.yml:23]:
```suggestion
          ./tools/run_command.py test \
```

File: .github/workflows/integration_test.yml
Comment (HakkyuKim) [reply .github/workflows/integration_test.yml:19]: I'm fine either way, as long as all workflow scripts are consistent: https://github.com/flutter-tizen/plugins/blob/master/.github/workflows/build.yml#L69.

@bbrto21

File: .github/workflows/integration_test.yml
Comment (HakkyuKim) [reply .github/workflows/integration_test.yml:23]: Ditto.

https://github.com/flutter-tizen/plugins/blob/master/.github/workflows/build.yml#L74

@bbrto21

File: .github/workflows/integration_test.yml
Comment (WonyoungChoi) [inline .github/workflows/integration_test.yml:9]: is this correct job name?

File: .github/workflows/integration_test.yml
Comment (WonyoungChoi) [inline .github/workflows/integration_test.yml:11]: Is there any reason to check `repository_onwer` ?

File: .github/workflows/integration_test.yml
Comment (HakkyuKim) [reply .github/workflows/integration_test.yml:9]: I named it so the information is sufficient in the UI, but I'm open to other suggestions.
Screenshot from 2021-09-28 15-47-56

File: .github/workflows/integration_test.yml
Comment (HakkyuKim) [reply .github/workflows/integration_test.yml:11]: I added it to block forked repository from triggering it.

File: .github/workflows/integration_test.yml
Comment (WonyoungChoi) [reply .github/workflows/integration_test.yml:9]: No problem. I was just curious. So, this means "test the changed packages"?

Comment (HakkyuKim) [reply .github/workflows/integration_test.yml:9]: Yes, that's correct.

File: .github/workflows/integration_test.yml
Comment (swift-kim) [reply .github/workflows/integration_test.yml:23]: As far as I remember the `pwd` in the for loop has a reason. Without `pwd`, the loop will fail due to the presence of `cd $d`.

Comment (HakkyuKim) [reply .github/workflows/integration_test.yml:23]: Ah yes, realized it a few minutes ago and deleted the link.

File: .github/workflows/integration_test.yml
Comment (bbrto21) [reply .github/workflows/integration_test.yml:19]: it seems nice

**Comment by HakkyuKim** [comment]
Should workflow be merged after setting multiple self-hosted runners with different device profiles? If this PR is merged, only a single self-hosted runner will be used for CI, testing plugin integration test on wearable-5.5 emulator.

#229 and #232 are required to run multiple self-hosted runners, running on tv emulators needs extra implementation.

**Review by bbrto21 — APPROVED** [review]
LGTM
I think it's meaningful even if it test plugins only on wearable emulator.
and if we use this, we may find new issues that we didn't expect.

**Comment by HakkyuKim** [comment]
Let's merge #229 first so we can add the `--platforms` options in this PR.

**Comment by HakkyuKim** [comment]
Rebased with current master to use the `--platforms` option.
The only thing changed since the latest review is:
- Use the `--platforms` option in workflows script: https://github.com/flutter-tizen/plugins/pull/169/commits/15dda0c3d93b51435dac16011a13bc5c45730a49
- Fix misused package name: https://github.com/flutter-tizen/plugins/pull/169/commits/1e75e08a41a5f8d076375cd7c9c928f1444b12b9

**Pull request merged**
